dango/dex/src/execute.rs

```diff
@@ -438,12 +440,21 @@ pub fn cron_execute(ctx: SudoCtx) -> anyhow::Result<Response> {
     let mut refunds = BTreeMap::new();
     let mut volumes = HashMap::new();
     let mut volumes_by_username = HashMap::new();
+    let mut collected_fees = Coins::new();
+    let mut fee_payments = BTreeMap::new();
 
     // Collect incoming orders and clear the temporary storage.
     let incoming_orders = INCOMING_ORDERS.drain(ctx.storage, None, None)?;
 
     // Add incoming orders to the persistent storage.
     for (order_key, order) in incoming_orders.values() {
+        // Ensure that the order was created in the current block. This should
+        // always be the case, or something has gone wrong.
+        ensure!(
+            order.created_at_block_height == ctx.block.height,
+            "incoming order was created in a previous block"
+        );
```

Comment:
I think `debug_assert` should be good enough. No need to run this in production.